### DISCUSSION TITLE
Unable to change power mode settings.

### DISCUSSION THREAD

**Discussion by Niuj79** [Seagate/openSeaChest#83] [Q&A]
I have two Seagate SkyHawk ST2000VX008 hard drives, and i cannot change anything in power modes. According to --showEPCSettings, non of the modes can be changed or saved. How to unlock these settings?
Disk are working under Debian, kernel version 5.10

> root@xxxxxxxxx~# openSeaChest_PowerControl -d /dev/sde --showEPCSettings
==========================================================================================
 openSeaChest_PowerControl - openSeaChest drive utilities - NVMe Enabled
 Copyright (c) 2014-2018 Seagate Technology LLC and/or its Affiliates, All Rights Reserved
 openSeaChest_PowerControl Version: 1.10.0-1_19_0 X86_64
 Build Date: Apr 23 2022
 Today: Sat Apr 23 16:26:21 2022
==========================================================================================

>/dev/sg4 - ST2000VX008-2E3164 - Z52A645Q - ATA
.

>===EPC Settings===
	* = timer is enabled
	C column = Changeable
	S column = Saveable
	All times are in 100 milliseconds

>Name       Current Timer Default Timer Saved Timer   Recovery Time C S
Idle A             *100          *100          *100          70                                  N N
Idle B                  0             0             0            220                                   N N
Idle C                  0             0             0            2300                                 N N
Standby Z  *18000         0             0            4800                                     N N

**Comment by vonericsen** [comment]
Hi @Niuj79,

What is the output from the `-i` option and the `--showAPMLevel` option?

**Reply by Niuj79** [reply]
Hi @vonericsen 

Output from `--showAPMelevel` is


>/dev/sg4 - ST2000VX008-2E3164 - Z52A645Q - ATA
>Showing APM Level is not supported on this device.

Output from `-i` is

> /dev/sg4 - ST2000VX008-2E3164 - Z52A645Q - ATA
	Model Number: ST2000VX008-2E3164
	Serial Number: Z52A645Q
	Firmware Revision: CV12
	World Wide Name: 5000C500C62A4256
	Drive Capacity (TB/TiB): 2.00/1.82
	Native Drive Capacity (TB/TiB): 2.00/1.82
	Temperature Data:
		Current Temperature (C): 29
		Highest Temperature (C): 30
		Lowest Temperature (C): 16
	Power On Time:  102 days 13 hours 
	Power On Hours: 8605.00
	MaxLBA: 3907029167
	Native MaxLBA: 3907029167
	Logical Sector Size (B): 512
	Physical Sector Size (B): 4096
	Sector Alignment: 0
	Rotation Rate (RPM): 5900
	Form Factor (inch): 3.5
	Last DST information:
		DST has never been run
	Long Drive Self Test Time:  4 hours 14 minutes 
	Interface speed:
		Max Speed (Gb/s): 6.0
		Negotiated Speed (Gb/s): 6.0
	Annualized Workload Rate (TB/yr): Not Reported
	Total Bytes Read (B): Not Reported
	Total Bytes Written (B): Not Reported
	Encryption Support: Not Supported
	Cache Size (MiB): 64.00
	Read Look-Ahead: Enabled
	Write Cache: Disabled
	Low Current Spinup: Disabled
	SMART Status: Good
	ATA Security Information: Supported
	Firmware Download Support: Full, Segmented
	Specifications Supported:
		ACS-2
		ATA8-ACS
		ATA/ATAPI-7
		ATA/ATAPI-6
		ATA/ATAPI-5
		ATA/ATAPI-4
		SATA 3.1
		SATA 3.0
		SATA 2.6
		SATA 2.5
		SATA II: Extensions
		SATA 1.0a
	Features Supported:
		SATA NCQ
		SATA Software Settings Preservation [Enabled]
		SATA Device Initiated Power Management
		HPA
		Power Management
		Security
		SMART [Enabled]
		DCO
		48bit Address
		PUIS [Enabled]
		GPL
		Streaming
		SMART Self-Test
		SMART Error Logging
		Write-Read-Verify
		EPC [Enabled]
		Sense Data Reporting [Enabled]
		SCT Error Recovery Control
		SCT Feature Control
		SCT Data Tables
		Host Logging
		Seagate In Drive Diagnostics (IDD)

**Comment by vonericsen** [comment]
Hi @Niuj79,

This took a bit of searching and asking questions to figure out.
The drive in your case is working according to the T13 (ATA) specs for the EPC feature, but it is saying what the timers are, just not allowing changes. The spec allows a device to choose whether the timers are changeable or not and in this case, this drive does not allow making any changes.

This is enforced by the firmware on the drive, so I asked around and it sounds like on Skyhawk drives this is intentional to not allow changes to the timers. I was able to confirm this on a Skyhawk drive that I tracked down. On the IronWolf and Exos drives I tested these were set to be changeable. Please be aware that I often work with engineering samples so they may not reflect final firmware configurations, although we do our best to match customer firmware as often as we can.

Unfortunately in your case, there is not a way to make these changeable since it is set by the device firmware to restrict this behavior in this configuration.
